### ISSUE TITLE
[a11y support] VA Chatbot: Prescription Skill Feature

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#65945]

**Comment by sara-amanda** [comment]
## CAIA Initial Video Demo Feedback for Chatbot
The following feedback is based on the [video provided by Matt. ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64157#issuecomment-1727905637) A link to staging and login credentials will need to be provided to allow for deeper and further testing. 

### Jamie's Feedback
@coforma-jamie 
- Click to sign-in > goes to the sign-in page —— need to make sure focus goes to the right place.
- Where should focus go AFTER sign in?
- What’s that long string of letters before “You are now in the Prescriptions Bot”? 
- Wouldn’t a screen reader read that out loud?
- Light gray font color (next to “Instructions”, for example).
- Probably doesn’t meet color contrast
- Prescription names in capital letters, can this is sentence case?
- List Prescriptions, Request Refills etc should be sentence case?
- “First, you have…. On 9/9/23” write out “September 9, 2023" for screen reader, and to match other dates in the chatbot?
- “Prescription status…” etc, is that a table, a DL, what? If a table, does it have a hidden <caption>

### SK's Feedback
@SarahKay8 
- I’m also worried about the gray color on the gray background. (Instructions section)
- Why does signing in trigger a long string of text, strange?
- (VA Chatbot as whole just because )—> chatbot’s automated and it's sending multiple messages within seconds of each other, meaning there wouldn’t be time for each message to announce if every message was set to announce immediately. (I don't think we’re able to queue up the messages with any dynamic announcement because there’s no way to know how long each individual user’s device will take to announce each message.)
- Sooo we should -
- Move focus to the first new message that appears (this will cause it to announce and place the user in a natural place to continue reading the rest of the new messages)
- The user can navigate forwards to the other messages at their own pace
- Once all the new messages have been read they will be back on the text input to send a reply.  I’ll watch again and add anything I may have missed but this is what I have for now. 

@coforma-terry

**Comment by coforma-terry** [comment]
Hey @nathalierayter @Joy-Elizabeth @mterwilliger - just following up to see if you're able to provide us with the requested info [above ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/65945#issuecomment-1731568653)ref staging credentials and the staging artifact?

Thanks!

**Comment by coforma-jamie** [comment]
## CAIA A11y Review of Chatbot on dev.va.gov

### Jamie's notes
#### Before login
- As @SarahKay8 suspected, the messages come in very quickly, the screen reader only has time to read the last message.
- When the user types in "prescriptions" and they're asked if they want to sign in, there's no spoken indication that there are "yes / no" buttons. The user can **type** "yes" or "no, though. You might want to have some SR-only text to let the user know that there are buttons to choose from, or explicitly say that the user can type in their response?

[Examples in video](https://github.com/department-of-veterans-affairs/va.gov-team/assets/135633989/f4da8019-eed4-4c01-bf68-02a08e4a7252)

**Comment by sara-amanda** [comment]
## ⚠️ CAIA Update: Launch-Blocking Issue

@coforma-jamie @sara-amanda @SarahKay8 met to discuss the following issue, and we have concluded the following @coforma-terry @nathalierayter @Joy-Elizabeth @mterwilliger:

### a11y-defect-1 
#### Critical. ❗️ Must be fixed before launch 
- These issues have the potential to severely disrupt the user experience.
- **This must be fixed before the Platform will certify a staging accessibility review**.

### VPAT Checked

- We located the VPAT and reviewed it. 
- The VPAT for this product only tests up to AA, not AAA, but this issue is `launch-blocking issue`. 

### What is Happening
Because this is an automated Chatbot sending multiple messages within seconds of each other, meaning there would not be time for each message to announce, in a natural place, if every message was set to do immediately. 

### Recommended Fix
- Move focus to the first message that appears. 
- The user can navigate forward to the new messages at their own pace. 
- Once all the new messages have been read, they would be back on the text input to reply. 

**Note**: The use of `aria-live="polite"` is well done in this Chatbot. 

<google-sheets-html-origin style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0);">

Topic | WCAG | COGA | VPAT Status
-- | -- | -- | --
Use a conversation pace that lets the user complete the task efficiently but doesn't rush them | WCAG 3.2.5 Change on Request | COGA 4.5.1 Ensure Controls and Content Do Not Move Unexpectedly | Not evaluated (AAA)

## W3C WCAG COGA Recommendations

> The WCAG document does not specifically address the wide range of cognitive abilities in target users. Latest recommendations from W3C’s Cognitive and Learning Disabilities Accessibility Task Force (COGA TF) have information explicitly directed at cognitive disabilities that can fill gaps in WCAG guidance. Ultimately, user tests will reveal how accessible the chatbot is.

### WCAG 3.2.5
- [Changes of context](https://www.w3.org/TR/WCAG22/#dfn-change-of-context) are initiated only by user request or a [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to turn off such changes.
- [Understanding Doc: SC 3.2.5: Change on Request (Level AAA)](https://www.w3.org/WAI/WCAG22/Understanding/change-on-request.html)
### COGA 4.5.1
- **When pacing chatbot messages in a sequence of multiple messages** (4.3.21), refer to COGA Design Pattern 4.5.1, which suggests not moving content unless initiated by the user. 
- **To conform with the design pattern**, consider waiting for the user to press a button to indicate they are done reading the current message before displaying the next one; or make sure the scrollable area does not move and interrupt the user’s reading as subsequent messages are received.
- [Read more about 4.5.1: making content usable for people with cognitive and learning disabilities](https://www.w3.org/TR/coga-usable/#ensure-controls-and-content-do-not-move-unexpectedly-pattern)

**Comment by nathalierayter** [comment]
Hi @sara-amanda @coforma-jamie @SarahKay8 thanks for this feedback; the conversation pace concerns that you raise are specific to the Rx feature (microservice), because the conversation pace is controlled by the chat platform (Power Virtual Agent, a Microsoft platform) and all that you reviewed should match the current state of the [in-production Chatbot](https://www.va.gov/contact-us/virtual-agent/) on VA.gov.

This is a COTS issue that my tech lead @ksk385 says that we can investigate, but is ultimately on Microsoft to solve. However, it is not insurmountable, as the user can scroll up at any time. We believe this should not block release of the Prescription feature, as the conversation pace is not unique to this feature and is instead a larger platform issue to be investigated.

**Comment by nathalierayter** [comment]
Thanks @coforma-jamie that's accurate and being addressed already; @mterwilliger our design researcher has additional testing specifically with AT users in flow right as of this week and can share the documentation around our aims with this round

**Comment by mterwilliger** [comment]
Thanks for the consideration @coforma-jamie, this is certainly a group we're prioritizing in this current round of research. The screener and facilitation guide for this specific round of research is now uploaded [here](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/virtual-agent/research/prescriptions) as well if you'd like to take a look. Although, I will add this facilitation guide will change a bit based on a few difficulties we identified in our first test session today. 

CC: @nathalierayter

**Comment by coforma-jamie** [comment]
@mterwilliger very happy to see the call for 4 veterans that are visually impaired and use AT, and the strong focus on accessibility in the facilitation guide! Feel free to reach out to the CAIA team about those difficulties, perhaps we can help prepare you for your next session? cc @sara-amanda @SarahKay8

**Comment by mterwilliger** [comment]
Thank you @coforma-jamie! I'll certainly reach out to the CAIA team if needed but I think I have a much better understanding after the initial session and can pivot based on that now.

**Comment by sara-amanda** [comment]
Closing ticket - deliverable shared
